app/controllers/types_controller.rb

```diff
@@ -0,0 +1,63 @@
+class TypesController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+  before_action :set_type, only: [:show, :edit, :update, :destroy]
+  respond_to :js, :html
+
+  def index
+    @types = current_user.types.order('created_at DESC').page(params[:page]).per(4)
+
+    @types = if params[:show_all] == 'true'
+               Type.all.order('created_at DESC').page(params[:page]).per(4)
+             else
+               current_user.types.order('created_at DESC').page(params[:page]).per(4)
+             end
+  end
+
+  def show
+  end
+
+  def new
+    @type = Type.new
+    @type.user_id = current_user.id
+  end
+
+  def edit
+  end
+
+  def create
+    @type = Type.new(type_params)
+    @type.user_id = current_user.id
+    render :new if @type.save != true
+  end
+
+  def update
+    render :edit if @type.update(type_params_update) != true
+  end
+
+  def destroy
+    @type.destroy
+  end
+
+  private
+
+  def set_type
+    @type = Type.find(params[:id])
+  end
+
+  def type_params
+    params.require(:type).permit(:name, :image, :user_id)
```

Comment:
@VanjaJeftic do you need user_id as a param? I don't think you are using it at all.